packages/multicast_dns/lib/multicast_dns.dart

```diff
@@ -219,11 +190,12 @@ class MDnsClient {
     final Stream<T> results = _resolver.addPendingRequest<T>(
         query.resourceRecordType, query.fullyQualifiedName, timeout);
 
-    // Send the request on all interfaces.
-    final List<int> packet = query.encode();
-    for (final RawDatagramSocket socket in _sockets) {
-      socket.send(packet, _mDnsAddress!, selectedMDnsPort);
+    if (_incomingIPv4 == null) {
+      throw StateError('Unable to send packet on null socket.');
```

Comment:
Won't this cause an exception to be thrown if the connection is over IPv6? 

For IPv6 in the old code, we'd just have no sockets to send over and not send anything, but now we'll throw an exception. I think you can just drop this check and everything should be fine, since you're using a null-safe dereference on the `send` call below.

Reply:
I'm curious why we didn't use equivalent logic for the IPv6 approach -- i.e. we never sent on the IPv6 equivalent of 0.0.0.0, only on the sockets bound to the individual network interfaces. I'll do a little digging to see if I can understand why.

Reply:
> Won't this cause an exception to be thrown if the connection is over IPv6?

correct!

> I think you can just drop this check and everything should be fine

Removed. I did not go deeper on the `IPV6` topic, I saw other MDNs packages (like [Hashicorp multicast mdns package](https://github.com/hashicorp/mdns/blob/e208a109b48c6cd841139c6913bddf0d7bb27a51/client.go#L184)) that they handle also `multicast` with `IPV6` but I did not want to go deeper in this PR. Maybe it's worth to open another issue for implementing it (or have a look if it's doable or not), what do you think? 🤔

Reply:
I guess this thread can be closed, right?

Reply:
I'm curious if you've had a chance to test this with IPv6. My worry is that the existing code was broken for IPv4 (and this fixes it), but this patch effectively removes IPv6 support altogether. That seems fine if it was previously broken, but if not, it's a regression for those users.

Other that phone users connecting over a cell connection, I suspect people with an IPv6-only connection are pretty few and far between so possibly not the end of the world, but still a regression.

Reply:
On balance, landing this is probably still a net win, but probably worth at least documenting the change.

Reply:
Thinking about it a little more I think there are two options here:
1. We add back the `_sockets` List (maybe as `_ipv6InterfaceSockets`) and the code that configures them as multicast -- but only when an `InternetAddress.anyIPv6` is passed to the constructor, and then the `send` code at the bottom similarly would either send on `_incomingIPv4` for IPv4, or the collection of IPv6 sockets for IPv6.
2. We update the constructor to specifically throw (maybe UnimplementedError) if `InternetAddress.anyIPv6` is passed in and document that IPv6 support has been removed, in the class/package docs.

Sorry to spot this at this point.

Reply:
Thanks again for the time spent on the review.

> Sorry to spot this at this point.

No need to say sorry, I'm just a guest here 😅 

I applied the changes, I put back `IPv6` logic and I updated tests accordingly.

(at this point, it's does not make sense to remove `IPv6` support.)